src/components/TodoApp.tsx

```diff
@@ -0,0 +1,133 @@
+import { useContext, useEffect, useState } from 'react';
+
+import { TodosContext } from '../TodosContext';
+import TodoList from './TodoList';
+import TodosFilter from './TodosFilter';
+import { Todo } from '../types/Todo';
+import { Status } from '../enums/Status';
+
+const filteredTodos = (todos: Todo[], filter: Status): Todo[] => {
+  switch (filter) {
+    case Status.Active:
+      return todos.filter(todo => !todo.completed);
+    case Status.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
+
+const TodoApp: React.FC = () => {
+  const { state, dispatch } = useContext(TodosContext);
+
+  const [title, setTitle] = useState('');
+  const [completedCount, setCompletedCount] = useState(0);
+  const [notCompletedCount, setNotCompletedCount] = useState(0);
+  const [isAllCompleted, setIsAllCompleted] = useState(false);
+  const [filter, setFilter] = useState(Status.All);
+
+  useEffect(() => {
+    const notCompleted = state.filter(todo => !todo.completed).length;
+    const completed = state.length - notCompleted;
+
+    if (completed === state.length) {
+      setIsAllCompleted(true);
+    } else {
+      setIsAllCompleted(false);
+    }
+
+    setNotCompletedCount(notCompleted);
+    setCompletedCount(completed);
+  }, [state, isAllCompleted]);
```

Comment:
That looks strange, simplify as I mentioned, just use variables for dynamic values such as completed and etc